tools/clang/lib/SPIRV/ModuleBuilder.cpp

```diff
@@ -613,6 +613,14 @@ uint32_t ModuleBuilder::getArrayType(uint32_t elemType, uint32_t count,
   return typeId;
 }
 
+uint32_t ModuleBuilder::getRuntimeArrayType(uint32_t elemType,
+                                            Type::DecorationSet decorations) {
+  const Type *type = Type::getRuntimeArray(theContext, elemType, decorations);
+  const uint32_t typeId = theContext.getResultIdForType(type);
+  theModule.addType(type, typeId);
```

Comment:
I think the default generated name from SPIRV-Tools for it (`_runtimearr_*`) is clear enough, so no need to emit more debug instructions. IMHO, we typically want to emit debug names if a) the user supplied an identifier for it in the source code, or b) it's too difficult to understand the default generated name.